tracing/tracer.go

```diff
@@ -0,0 +1,103 @@
+package tracing
+
+import (
+	"fmt"
+
+	"github.com/opentracing/opentracing-go"
+	zipkinot "github.com/openzipkin-contrib/zipkin-go-opentracing"
+	zipkin "github.com/openzipkin/zipkin-go"
+	"github.com/openzipkin/zipkin-go/reporter"
+	zipkinhttp "github.com/openzipkin/zipkin-go/reporter/http"
+	"github.com/reddit/baseplate.go/log"
+)
+
+var zipkinReporter reporter.Reporter
+
+// Tracer is exported to support start tracible server etc
+var Tracer = opentracing.GlobalTracer()
+
+// InitTracer configure the global tracer
+func InitTracer(
+	serviceName, addr string,
+	sampleRate float64,
+	logger log.Wrapper,
+) {
+	if addr == "" {
+		logger("Endpoint to send spans to is undefined")
+		return
+	}
+
+	zipkinReporter := zipkinhttp.NewReporter(addr)
+
+	endpoint, err := zipkin.NewEndpoint(serviceName, "0.0.0.0:0")
+	if err != nil && logger != nil {
+		logger(fmt.Sprintf("Unable to create local endpoint: %+v\n", err))
+	}
+
+	sampler, err := zipkin.NewCountingSampler(sampleRate)
+	if err != nil && logger != nil {
+		logger(fmt.Sprintf("Unable to create sampler: %+v\n", err))
+	}
+
+	nativeTracer, err := zipkin.NewTracer(
+		zipkinReporter,
+		zipkin.WithSampler(sampler),
+		zipkin.WithLocalEndpoint(endpoint),
+	)
+
+	tracer := zipkinot.Wrap(nativeTracer)
+	opentracing.SetGlobalTracer(tracer)
+
+	if logger != nil {
+		logger("Global tracer created")
+	}
```

Comment:
I don't think this log is necessary. As a principal of library and logging, library code should only do logging when it's something bad and it cannot return the error to the caller (in most cases, it should try avoid logging as best as possible, only return the error to the caller and let the caller to decide how to handle that).

Also the other logger usages should have the nil check :)